frontend/packages/console-shared/src/utils/label-filter.ts

```diff
@@ -0,0 +1,14 @@
+import * as _ from 'lodash';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { toRequirements } from '@console/internal/module/k8s/selector';
+import { requirementToString } from '@console/internal/module/k8s/selector-requirement';
+
+export const getLabelsAsString = (obj: K8sResourceCommon): string[] => {
+  const { labels } = obj.metadata;
+  const requirements = toRequirements(labels);
+  const flatLabels: Set<string> = requirements.reduce((acc, requirement) => {
+    acc.add(requirementToString(requirement));
+    return acc;
+  }, new Set<string>());
+  return [...flatLabels];
```

Comment:
`Set` is not needed here since you should never have duplicates and it's only a single resource.